WooCommerce/WooCommerceTests/ViewRelated/HubMenu/HubMenuViewModelTests.swift

```diff
@@ -568,7 +568,7 @@ final class HubMenuViewModelTests: XCTestCase {
     @MainActor
     func test_navigateToDestination_replaces_navigationPath_with_specified_destination() throws {
         // Given
-        let generalAppSettings = try mockGeneralAppSettingsStorage(isInAppPurchaseEnabled: true)
+        let generalAppSettings = try mockGeneralAppSettingsStorage()
```

Comment:
Do you still need to pass a mock?

It's fine, arguably good practice... just wondering if the mock is changing the behaviour at all.

Reply:
It doesn't seem really needed, by not injecting the mock explicitly the test still passes as uses the Servicelocator's instance as default. It seems a bit better though, as the mock relies on `MockInMemoryStorage` as well when saving settings. 

I personally don't like hiding the dependencies much if can be avoided, is not the first time I found issues with tests not working as expected because were relying on a ServiceLocator as default. I'll leave the mock for now, but I think it could benefit from some refactoring around the init at some point the next time we work in the `HubMenuViewModel` 👍